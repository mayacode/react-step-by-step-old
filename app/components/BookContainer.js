import React, { Component } from 'react';
import BookSearch from './BookSearch';
import BookList from './BookList';

const googleBooksSearchAPI = 'https://www.googleapis.com/books/v1/volumes?q=';

export default class BookContainer extends Component {
    state = {
        keyword: '',
        books: []
    };

    componentDidUpdate = (prevProps, prevState) => {
        const { keyword } = this.state;
        if (keyword.length > 2 && prevState.keyword !== keyword) {
            fetch(`${googleBooksSearchAPI}${keyword}`)
                .then(response => response.json())
                .then(myJson => {
                    this.setState({ books: myJson.items });
                });
        }
    };

    changeHandler = e => {
        const keyword = e.target.value;
        this.setState({ keyword });
    };

    render() {
        return (
            <section>
                <BookSearch changeHandler={this.changeHandler} />
                <BookList books={this.state.books}/>
            </section>
        );
    }
}
