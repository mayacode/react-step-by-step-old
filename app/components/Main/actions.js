export const clickedMyButton = () => ({
  type: 'CLICKED',
});
