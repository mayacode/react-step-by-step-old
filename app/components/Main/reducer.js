const initialStore = {
  clicks: 0,
};

const myButtonReducer = (store = initialStore, action = {}) => {
  switch (action.type) {
    case 'CLICKED':
      return { clicks: store.clicks + 1 };
    default:
      return store;
  }
};

export default myButtonReducer;
