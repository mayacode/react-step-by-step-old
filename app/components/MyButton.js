import React, { Fragment } from 'react';

export default ({clickHandler, number}) => (
  <Fragment>
    <button onClick={clickHandler}>Click me!</button>
        You clicked me {number} times
  </Fragment>
);
