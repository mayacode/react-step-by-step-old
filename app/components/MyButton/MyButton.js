import React, { Fragment } from 'react';
import { func, number } from 'prop-types';

const MyButton = ({clickHandler, clicks}) => (
  <Fragment>
    <button onClick={clickHandler}>Click me!</button>
    You clicked me {clicks} times
  </Fragment>
);

MyButton.displayName = 'MyButton';
MyButton.propTypes = {
  clickHandler: func.isRequired,
  clicks: number.isRequired,
};

export default MyButton;
