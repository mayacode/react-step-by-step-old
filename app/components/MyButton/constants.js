export const MY_BUTTON_CLICKED = 'MY_BUTTON_CLICKED';