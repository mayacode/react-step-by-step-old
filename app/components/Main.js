import React, { Component } from 'react';
import MyButton from './MyButton';

export default class Main extends Component {
    state = {
        number: 0,
    };

    clickHandler = () => {
        const num = this.state.number;
        this.setState({ number: num + 1 });
    };

    render() {
        return (
            <div>
                Hello World!
                <MyButton number={this.state.number} clickHandler={this.clickHandler} />
            </div>
        );
    }
}
