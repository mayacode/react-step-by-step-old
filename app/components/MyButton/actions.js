import { MY_BUTTON_CLICKED } from './constants';

export const clickedMyButton = () => ({
  type: MY_BUTTON_CLICKED,
});