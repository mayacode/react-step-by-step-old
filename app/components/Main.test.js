import React from 'react';
import { shallow } from 'enzyme';

import Main from './Main';
import MyButton from './MyButton/container';
import BookContainer from './BookContainer';


describe('<Main />', () => {
  let wrapper;
  let myButton;
  let bookContainer;

  beforeAll(() => {
    wrapper = shallow(<Main />);
    myButton = wrapper.find(MyButton);
    bookContainer = wrapper.find(BookContainer)
  });

  it('matches a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('<MyButton />', () => {
    it('is only 1', () => {
      expect(myButton.length).toEqual(1);
    });
  });



  describe('<BookContainer />', () => {
    it('is only 1', () => {
      expect(bookContainer.length).toEqual(1);
    });
  });
});
