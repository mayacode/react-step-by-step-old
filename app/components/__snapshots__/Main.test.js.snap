// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> matches a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Hello World!",
        <Connect(MyButton) />,
        <hr />,
        <BookContainer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Hello World!",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Hello World!",
          <Connect(MyButton) />,
          <hr />,
          <BookContainer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Hello World!",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<MyButton /> matches a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Hello World!",
        <Connect(MyButton) />,
        <hr />,
        <BookContainer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Hello World!",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Hello World!",
          <Connect(MyButton) />,
          <hr />,
          <BookContainer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Hello World!",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
